// @strapi/design-system/dist/Field/FieldInput.js
var F = Object.defineProperty;
var a = Object.getOwnPropertySymbols;
var g = Object.prototype.hasOwnProperty, b = Object.prototype.propertyIsEnumerable;
var f = (e, o, r) => o in e ? F(e, o, { enumerable: !0, configurable: !0, writable: !0, value: r }) : e[o] = r, h = (e, o) => {
  for (var r in o || (o = {}))
    g.call(o, r) && f(e, r, o[r]);
  if (a)
    for (var r of a(o))
      b.call(o, r) && f(e, r, o[r]);
  return e;
};
var $ = (e, o) => {
  var r = {};
  for (var t in e)
    g.call(e, t) && o.indexOf(t) < 0 && (r[t] = e[t]);
  if (e != null && a)
    for (var t of a(e))
      o.indexOf(t) < 0 && b.call(e, t) && (r[t] = e[t]);
  return r;
};
import i, { forwardRef as B } from "react";
import R from "styled-components";
import n from "prop-types";
import { sizes as D } from "../themes/sizes.js";
import { inputFocusStyle as O } from "../themes/utils.js";
import { useField as z } from "./FieldContext.js";
import { Flex as A } from "../Flex/Flex.js";
import { Box as y } from "../Box/Box.js";
const w = {
  S: 6.5,
  M: 10.5
}, L = R.input`
  border: none;
  border-radius: ${({ theme: e }) => e.borderRadius};
  padding-bottom: ${({ size: e }) => `${w[e] / 16}rem`};
  padding-left: ${({ theme: e, hasLeftAction: o }) => o ? 0 : e.spaces[4]};
  padding-right: ${({ theme: e, hasRightAction: o }) => o ? 0 : e.spaces[4]};
  padding-top: ${({ size: e }) => `${w[e] / 16}rem`};
  cursor: ${(e) => e["aria-disabled"] ? "not-allowed" : void 0};

  color: ${({ theme: e }) => e.colors.neutral800};
  font-weight: 400;
  // TODO: Make sure to use the theme when it's ready
  font-size: ${14 / 16}rem;
  display: block;
  width: 100%;
  background: inherit;

  ::placeholder {
    color: ${({ theme: e }) => e.colors.neutral500};
    opacity: 1;
  }

  &[aria-disabled='true'] {
    color: inherit;
  }

  //focus managed by InputWrapper
  &:focus {
    outline: none;
    box-shadow: none;
  }
`, M = R(A)`
  border: 1px solid ${({ theme: e, hasError: o }) => o ? e.colors.danger600 : e.colors.neutral200};
  border-radius: ${({ theme: e }) => e.borderRadius};
  background: ${({ theme: e }) => e.colors.neutral0};
  ${O()}

  ${({ theme: e, disabled: o }) => o ? `
    color: ${e.colors.neutral600};
    background: ${e.colors.neutral150};
  
  ` : void 0}
`, l = B((P, I) => {
  var c = P, { endAction: e, startAction: o, disabled: r, onChange: t, size: p } = c, k = $(c, ["endAction", "startAction", "disabled", "onChange", "size"]);
  const { id: d, error: u, hint: v, name: x } = z();
  let s;
  u ? s = `${d}-error` : v && (s = `${d}-hint`);
  const m = Boolean(u), C = (E) => {
    r || t(E);
  };
  return /* @__PURE__ */ i.createElement(M, {
    size: p,
    justifyContent: "space-between",
    hasError: m,
    disabled: r
  }, o && /* @__PURE__ */ i.createElement(y, {
    paddingLeft: 3,
    paddingRight: 2
  }, o), /* @__PURE__ */ i.createElement(L, h({
    id: d,
    name: x,
    ref: I,
    "aria-describedby": s,
    "aria-invalid": m,
    "aria-disabled": r,
    hasLeftAction: Boolean(o),
    hasRightAction: Boolean(e),
    onChange: C,
    size: p
  }, k)), e && /* @__PURE__ */ i.createElement(y, {
    paddingLeft: 2,
    paddingRight: 3
  }, e));
});
l.displayName = "FieldInput";
l.defaultProps = {
  disabled: !1,
  endAction: void 0,
  size: "M",
  startAction: void 0,
  onChange() {
  }
};
l.propTypes = {
  disabled: n.bool,
  endAction: n.element,
  onChange: n.func,
  size: n.oneOf(Object.keys(D.input)),
  startAction: n.element
};
export {
  l as FieldInput,
  M as InputWrapper
};
